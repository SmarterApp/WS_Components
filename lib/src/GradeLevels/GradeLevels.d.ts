export declare enum GradeLevels {
    NA = 0,
    Grade3 = 1,
    Grade4 = 2,
    Grade5 = 4,
    Grade6 = 8,
    Grade7 = 16,
    Grade8 = 32,
    Grade9 = 64,
    Grade10 = 128,
    Grade11 = 256,
    Grade12 = 512,
    Elementary = 7,
    Middle = 56,
    High = 960,
    All = 1023,
}
export declare class GradeLevel {
    static gradeCaseToString(grade: GradeLevels): string | undefined;
    static gradeCaseToShortString(grade: GradeLevels): string | undefined;
    static gradeLevelToString(grades: GradeLevels): string | undefined;
    static gradeLevelContains(haystack: GradeLevels, needle: GradeLevels): boolean;
    static stringToGradeLevel(gradeString: string): GradeLevels;
    static gradeLevelAdd(grade: GradeLevels, gradeString: string): number;
}
