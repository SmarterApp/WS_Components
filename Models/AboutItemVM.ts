import * as Rubric from '../PageTabs/Rubric'
import * as ItemCardViewModel from './ItemCardViewModel'
import * as API from '../Models/ApiModels'

export const ScoreSearchClient = (params: { bankKey: number, itemKey: number }) => API.get<AboutThisItem>("/api/aboutItem", params)

export interface AboutThisItem {
    rubrics: Rubric.Rubric[];
    itemCardViewModel: ItemCardViewModel.ItemCardViewModel;
    depthOfKnowledge: string;
    targetDescription: string;
    commonCoreStandardsDescription: string;
    educationalDifficulty: string;
    evidenceStatement: string;
}