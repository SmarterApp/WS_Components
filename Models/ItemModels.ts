import * as $ from 'jquery';
import { GradeLevels } from './GradeLevels';
import * as API from './ApiModels';
import * as ItemCardViewModel from './ItemCardViewModel';

export interface Subject {
    code: string;
    label: string;
}

export interface TechType extends Subject { }

export interface FilterOptions {
    subjects: Subject[];
    grades: GradeLevels[];
    techTypes: TechType[];
}

export interface ItemFilter extends FilterOptions { };