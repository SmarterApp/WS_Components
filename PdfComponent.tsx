import * as React from 'react';
import { ItemGroup } from '../Models';
import { ItemPage } from './ItemPage';
import { FirstPage } from "./FirstPage";

interface Props {
    items: ItemGroup[];
    grade: string;
    subject: string;
    pageBaseUrl: string;
    ivsBaseUrl: string;
    displayTitlePage: boolean;
}

export class PdfComponent extends React.Component<Props, {}> {
    render() {
        const itemPages = this.props.items.map(item => (
            <ItemPage itemData={item} key={item.questions[0].id}/>
        ));

        return (
            <html>
                <head>
                    <base href={this.props.pageBaseUrl} />
                    <link rel='stylesheet' href='css/pdf.css' />
                    <link rel='stylesheet' href={this.props.ivsBaseUrl + "/Shared/CSS/Universal/items.css"} />
                    <link rel='stylesheet' href={this.props.ivsBaseUrl + "/Projects/SBAC/css/modernItems.css"} />
                </head>
                <body>
                    {this.props.displayTitlePage 
                        ? <FirstPage grade={this.props.grade} subject={this.props.subject} /> 
                        : null}
                    {itemPages}
                </body>
            </html>
        );
    }
}

