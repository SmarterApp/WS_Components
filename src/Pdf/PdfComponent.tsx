import * as path from 'path';
import * as React from 'react';
import { ItemGroup } from '../Models';
import { ItemPage } from './ItemPage';
import { FirstPage } from "./FirstPage";


interface Props {
    items: ItemGroup[];
    grade: string;
    subject: string;
    ivsBaseUrl: string;
    cssUrl: string;
    displayTitlePage: boolean;
}

export class PdfComponent extends React.Component<Props, {}> {
    render() {
        const itemPages = this.props.items.map(item => (
            <ItemPage itemData={item} key={item.questions[0].id}/>
        ));

        return (
            <html>
                <head>
                    <meta charSet="UTF-8" />
                    <link rel='stylesheet' href={this.props.cssUrl} />
                    <link rel='stylesheet' href={this.props.ivsBaseUrl + "/Shared/CSS/Universal/items.css"} />
                    <link rel='stylesheet' href={this.props.ivsBaseUrl + "/Projects/SBAC/css/modernItems.css"} />
                </head>
                <body>
                    {this.props.displayTitlePage
                        ? <FirstPage grade={this.props.grade} subject={this.props.subject} />
                        : null}
                    {itemPages}
                </body>
            </html>
        );
    }
}
