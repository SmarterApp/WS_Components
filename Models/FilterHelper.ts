import { ItemCardViewModel } from "../Models/ItemCardViewModel";
import { ItemFilter, FilterOptions } from "../Models/ItemModels";
import * as GradeLevels from '../Models/GradeLevels';
import { parseQueryString } from "../Models/ApiModels";
import { AdvancedFilterCategory, AdvancedFilters, OptionType, AdvancedFilterOption } from "../filter/AdvancedFilterModel";

export class FilterHelper {

    //TODO: Get this from the server
    static getFilterOptions(): AdvancedFilters  {
        const subjectsFilterOptions: AdvancedFilterOption[] = [{
            label: "Mathematics",
            key: "MATH",
            isSelected: false,
            type: OptionType.button
        },
        {
            label: "English",
            key: "ELA",
            isSelected: false,
            type: OptionType.button
        }];

        const gradesFilterOptions: AdvancedFilterOption[] = [{
            label: GradeLevels.caseToString(GradeLevels.GradeLevels.Elementary),
            key: String(GradeLevels.GradeLevels.Elementary),
            isSelected: false,
            type: OptionType.button
        }, {
            label: GradeLevels.caseToString(GradeLevels.GradeLevels.Grade3),
            key: String(GradeLevels.GradeLevels.Grade3),
            isSelected: false,
            type: OptionType.button
        }, {
            label: GradeLevels.caseToString(GradeLevels.GradeLevels.Grade4),
            key: String(GradeLevels.GradeLevels.Grade4),
            isSelected: false,
            type: OptionType.button
        }, {
            label: GradeLevels.caseToString(GradeLevels.GradeLevels.Grade5),
            key: String(GradeLevels.GradeLevels.Grade5),
            isSelected: false,
            type: OptionType.button
        }, {
            label: GradeLevels.caseToString(GradeLevels.GradeLevels.Grade6),
            key: String(GradeLevels.GradeLevels.Grade6),
            isSelected: false,
            type: OptionType.button
        }, {
            label: GradeLevels.caseToString(GradeLevels.GradeLevels.Grade7),
            key: String(GradeLevels.GradeLevels.Grade7),
            isSelected: false,
            type: OptionType.button
        }, {
            label: GradeLevels.caseToString(GradeLevels.GradeLevels.Grade8),
            key: String(GradeLevels.GradeLevels.Grade8),
            isSelected: false,
            type: OptionType.button
        }, {
            label: GradeLevels.caseToString(GradeLevels.GradeLevels.High),
            key: String(GradeLevels.GradeLevels.High),
            isSelected: false,
            type: OptionType.button
        }];

        const techTypesFilterOptions: AdvancedFilterOption[] = [{
            label: "CAT",
            key: "CAT",
            isSelected: false,
            type: OptionType.button
        }, {
            label: "Performance Items",
            key: "PT",
            isSelected: false,
            type: OptionType.button
        }];

        const subjects: AdvancedFilterCategory = {
            disabled: false,
            isMultiSelect: false,
            label: "Subjects",
            helpText: "Subjects HelpText here.",
            filterOptions: [...subjectsFilterOptions],
            displayAllButton: true
        }

        const grades: AdvancedFilterCategory = {
            disabled: false,
            isMultiSelect: false,
            label: "Grade",
            helpText: "Grade HelpText here.",
            filterOptions: [...gradesFilterOptions],
            displayAllButton: true
        }

        const techTypes: AdvancedFilterCategory = {
            disabled: false,
            isMultiSelect: false,
            label: "TechType",
            helpText: "TechType HelpText here.",
            filterOptions: [...techTypesFilterOptions],
            displayAllButton: false
        }

        const value: AdvancedFilters = {subjects, grades, techTypes};
        
        return value;
    }

    static filter(itemCards: ItemCardViewModel[], filter: AdvancedFilterCategory[]): ItemCardViewModel[] {
        const grades = filter.find(afc => afc.label === "Grade");
        const subjects = filter.find(afc => afc.label === "Subjects");
        const techTypes = filter.find(afc => afc.label === "TechType");

        if (grades && grades.filterOptions) {
            let selectedGrades = GradeLevels.GradeLevels.NA;

            grades.filterOptions.forEach(gradeFilter => {
                if (gradeFilter.isSelected) {
                    selectedGrades = selectedGrades | Number(gradeFilter.key);
                }
            });

            if (selectedGrades === GradeLevels.GradeLevels.NA) {
                selectedGrades = GradeLevels.GradeLevels.All;
            }

            itemCards = itemCards.filter(g => GradeLevels.contains(selectedGrades, g.grade));
        }

        if (subjects && subjects.filterOptions) {
            let subjectCode:string|undefined;

            subjects.filterOptions.forEach(s => {
                if(s.isSelected){
                    subjectCode = s.key;
                }
            });
            if(subjectCode !== undefined){
                itemCards = itemCards.filter(i => subjectCode === i.subjectCode);
            }
        }
        //TODO: What is CAT technology? Filter? Ignore?
        if (techTypes && techTypes.filterOptions) {
            if (techTypes.filterOptions.find(t => t.key === "PT" && t.isSelected)) {
                itemCards = itemCards.filter(i => i.isPerformanceItem);
            } else if (techTypes.filterOptions.find(t => t.key === "CAT" && t.isSelected)) {
                itemCards = itemCards.filter(i => !i.isPerformanceItem);
            }
        }

        return itemCards;
    }

    static updateUrl(filter: AdvancedFilterCategory[]) {
        let pairs: string[] = [];

        const grades = filter.find(afc => afc.label === "Grade");
        const subjects = filter.find(afc => afc.label === "Subjects");
        const techTypes = filter.find(afc => afc.label === "TechType");

        if (grades && !grades.disabled) {
            const gradeString: string[] = [];

            grades.filterOptions.forEach(g => {
                if (g.isSelected) {
                    gradeString.push(g.key.toString());
                }
            });

            if (gradeString.length !== 0) {
                gradeString.join(',');
                pairs.push("gradeLevels=" + gradeString);
            }
        }
        if (subjects && !subjects.disabled) {
            const subjString: string[] = [];

            subjects.filterOptions.forEach(s => {
                if (s.isSelected) {
                    subjString.push(s.key);
                }
            });

            if (subjString.length !== 0) {
                subjString.join(',');
                pairs.push("subjects=" + subjString);
            }
        }
        if (techTypes && !techTypes.disabled) {
            const techTypesString: string[] = [];

            techTypes.filterOptions.forEach(t => {
                if (t.isSelected) {
                    techTypesString.push(t.key);
                }
            });
            if (techTypesString.length !== 0) {
                techTypesString.join(',');
                pairs.push("techTypes=" + techTypesString);
            }
        }

        let query: string;
        if (pairs.length === 0) {
            query = "/";
        } else {
            query = "?" + pairs.join("&");
        }
        history.replaceState(null, "", query);
    }

    static readUrl(filterOptions: AdvancedFilters) {
        const queryObject = parseQueryString(window.location.href);
        const subjects = queryObject["subjects"]
            ? queryObject["subjects"]!.map(
                subjCode => {
                    let subject = filterOptions.subjects.filterOptions.find(s => s.key === subjCode);
                    if (subject) {
                        return {
                            code: subject.key,
                            label: subject.label
                        }
                    }
                }) : [];

        const grades = queryObject["grades"]
            ? queryObject["grades"]!
                .map(gradeCode => Number(gradeCode) as GradeLevels.GradeLevels)
            : [];

        const techTypes = queryObject["techTypes"]
            ? queryObject["techTypes"]!.map(typeCode => {
                let techType = filterOptions.techTypes.filterOptions.find(t => t.key === typeCode);
                if (techType) {
                    return {
                        code: techType.key,
                        label: techType.label
                    }
                }
            }) : [];

        return {
            subjects: subjects,
            grades: grades,
            techTypes: techTypes
        } as ItemFilter;
    }
}