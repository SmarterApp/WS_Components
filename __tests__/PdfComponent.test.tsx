import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { PdfComponent } from '../PdfComponent';
import * as Models from "../../Models";
import { shallow } from 'enzyme';
import { GradeLevels } from '../../../../client/src/Models/GradeLevels';

const rubric:  Models.Rubric = {
    language: "english",
    rubricEntries: [{    
        scorepoint: "",
        name: "",
        value: ""
    }],
    samples: [{
        maxValue: "",
        minValue: "",
        sampleResponses: [{
            purpose: "",
            scorePoint: "",
            name: "",
            sampleContent: ""
        }]     
    }]
}

const itemVM: Models.ItemViewModel = {
    bankKey: 1,
    itemKey: 1,
    gradeLabel: "grade 3",
    subjectLabel: "math",
    claimCode: "234536",
    targetId: "23463467",
    domain: "MATH",
    depthOfKnowledge: "",
    commonCoreStandardId: "",
    title: "",
    grade: GradeLevels.Grade6,
    subjectCode: "MATH",
    claimLabel: "Math Claim",
    target: "1-3",
    interactionTypeCode: "",
    interactionTypeLabel: "",
    isPerformanceItem: false
}

const aboutItemVM: Models.AboutItemViewModel = {
    itemCardViewModel: itemVM,
    rubrics: [rubric],
    targetDescription: "string",
    depthOfKnowledge: "string",
    commonCoreStandardsDescription: "string",
    educationalDifficulty: "string",
    evidenceStatement: "string",
    associatedItems: "string"
}

const itemView: Models.ItemView = {
    id: "12334",
    html: undefined,
    picturePath: undefined,
    captured: true,
    type: Models.ViewType.html
}

const question: Models.Question = {
    id: "12334",
    view: itemView,
    data: aboutItemVM,
    questionNumber: 1
}

const itemGroup : Models.ItemGroup = {
    passage: undefined,
    questions: [question]
}

const props = {
    items:  [itemGroup],
    grade: "3",
    subject: "math",
    pageBaseUrl: "http://example.com",
    ivsBaseUrl: "http://example.com"
}

describe("PdfComponent", () => {
    it("matches snapshot", () => {
        expect(shallow(<PdfComponent {...props} />)).toMatchSnapshot();
    })
})