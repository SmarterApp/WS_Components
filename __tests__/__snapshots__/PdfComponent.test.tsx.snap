// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PdfComponent matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PdfComponent
    displayTitlePage={true}
    grade="3"
    items={
      Array [
        Object {
          "passage": undefined,
          "questions": Array [
            Object {
              "data": Object {
                "associatedItems": "string",
                "commonCoreStandardsDescription": "string",
                "depthOfKnowledge": "string",
                "educationalDifficulty": "string",
                "evidenceStatement": "string",
                "itemCardViewModel": Object {
                  "bankKey": 1,
                  "claimCode": "234536",
                  "claimLabel": "Math Claim",
                  "commonCoreStandardId": "",
                  "depthOfKnowledge": "",
                  "domain": "MATH",
                  "grade": 8,
                  "gradeLabel": "grade 3",
                  "interactionTypeCode": "",
                  "interactionTypeLabel": "",
                  "isPerformanceItem": false,
                  "itemKey": 1,
                  "subjectCode": "MATH",
                  "subjectLabel": "math",
                  "target": "1-3",
                  "targetId": "23463467",
                  "title": "",
                },
                "rubrics": Array [
                  Object {
                    "language": "english",
                    "rubricEntries": Array [
                      Object {
                        "name": "",
                        "scorepoint": "",
                        "value": "",
                      },
                    ],
                    "samples": Array [
                      Object {
                        "maxValue": "",
                        "minValue": "",
                        "sampleResponses": Array [
                          Object {
                            "name": "",
                            "purpose": "",
                            "sampleContent": "",
                            "scorePoint": "",
                          },
                        ],
                      },
                    ],
                  },
                ],
                "targetDescription": "string",
              },
              "id": "12334",
              "questionNumber": 1,
              "view": Object {
                "captured": true,
                "html": undefined,
                "id": "12334",
                "picturePath": undefined,
                "type": 1,
              },
            },
          ],
        },
      ]
    }
    ivsBaseUrl="http://example.com"
    pageBaseUrl="http://example.com"
    subject="math"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <head>
          <base
            href="http://example.com"
          />
          <meta
            charSet="UTF-8"
          />
          <link
            href="css/pdf.css"
            rel="stylesheet"
          />
          <link
            href="http://example.com/Shared/CSS/Universal/items.css"
            rel="stylesheet"
          />
          <link
            href="http://example.com/Projects/SBAC/css/modernItems.css"
            rel="stylesheet"
          />
        </head>,
        <body>
          <FirstPage
            grade="3"
            subject="math"
          />
          <ItemPage
            itemData={
              Object {
                "passage": undefined,
                "questions": Array [
                  Object {
                    "data": Object {
                      "associatedItems": "string",
                      "commonCoreStandardsDescription": "string",
                      "depthOfKnowledge": "string",
                      "educationalDifficulty": "string",
                      "evidenceStatement": "string",
                      "itemCardViewModel": Object {
                        "bankKey": 1,
                        "claimCode": "234536",
                        "claimLabel": "Math Claim",
                        "commonCoreStandardId": "",
                        "depthOfKnowledge": "",
                        "domain": "MATH",
                        "grade": 8,
                        "gradeLabel": "grade 3",
                        "interactionTypeCode": "",
                        "interactionTypeLabel": "",
                        "isPerformanceItem": false,
                        "itemKey": 1,
                        "subjectCode": "MATH",
                        "subjectLabel": "math",
                        "target": "1-3",
                        "targetId": "23463467",
                        "title": "",
                      },
                      "rubrics": Array [
                        Object {
                          "language": "english",
                          "rubricEntries": Array [
                            Object {
                              "name": "",
                              "scorepoint": "",
                              "value": "",
                            },
                          ],
                          "samples": Array [
                            Object {
                              "maxValue": "",
                              "minValue": "",
                              "sampleResponses": Array [
                                Object {
                                  "name": "",
                                  "purpose": "",
                                  "sampleContent": "",
                                  "scorePoint": "",
                                },
                              ],
                            },
                          ],
                        },
                      ],
                      "targetDescription": "string",
                    },
                    "id": "12334",
                    "questionNumber": 1,
                    "view": Object {
                      "captured": true,
                      "html": undefined,
                      "id": "12334",
                      "picturePath": undefined,
                      "type": 1,
                    },
                  },
                ],
              }
            }
          />
        </body>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <base
              href="http://example.com"
            />,
            <meta
              charSet="UTF-8"
            />,
            <link
              href="css/pdf.css"
              rel="stylesheet"
            />,
            <link
              href="http://example.com/Shared/CSS/Universal/items.css"
              rel="stylesheet"
            />,
            <link
              href="http://example.com/Projects/SBAC/css/modernItems.css"
              rel="stylesheet"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "href": "http://example.com",
            },
            "ref": null,
            "rendered": null,
            "type": "base",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "charSet": "UTF-8",
            },
            "ref": null,
            "rendered": null,
            "type": "meta",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "href": "css/pdf.css",
              "rel": "stylesheet",
            },
            "ref": null,
            "rendered": null,
            "type": "link",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "href": "http://example.com/Shared/CSS/Universal/items.css",
              "rel": "stylesheet",
            },
            "ref": null,
            "rendered": null,
            "type": "link",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "href": "http://example.com/Projects/SBAC/css/modernItems.css",
              "rel": "stylesheet",
            },
            "ref": null,
            "rendered": null,
            "type": "link",
          },
        ],
        "type": "head",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <FirstPage
              grade="3"
              subject="math"
            />,
            Array [
              <ItemPage
                itemData={
                  Object {
                    "passage": undefined,
                    "questions": Array [
                      Object {
                        "data": Object {
                          "associatedItems": "string",
                          "commonCoreStandardsDescription": "string",
                          "depthOfKnowledge": "string",
                          "educationalDifficulty": "string",
                          "evidenceStatement": "string",
                          "itemCardViewModel": Object {
                            "bankKey": 1,
                            "claimCode": "234536",
                            "claimLabel": "Math Claim",
                            "commonCoreStandardId": "",
                            "depthOfKnowledge": "",
                            "domain": "MATH",
                            "grade": 8,
                            "gradeLabel": "grade 3",
                            "interactionTypeCode": "",
                            "interactionTypeLabel": "",
                            "isPerformanceItem": false,
                            "itemKey": 1,
                            "subjectCode": "MATH",
                            "subjectLabel": "math",
                            "target": "1-3",
                            "targetId": "23463467",
                            "title": "",
                          },
                          "rubrics": Array [
                            Object {
                              "language": "english",
                              "rubricEntries": Array [
                                Object {
                                  "name": "",
                                  "scorepoint": "",
                                  "value": "",
                                },
                              ],
                              "samples": Array [
                                Object {
                                  "maxValue": "",
                                  "minValue": "",
                                  "sampleResponses": Array [
                                    Object {
                                      "name": "",
                                      "purpose": "",
                                      "sampleContent": "",
                                      "scorePoint": "",
                                    },
                                  ],
                                },
                              ],
                            },
                          ],
                          "targetDescription": "string",
                        },
                        "id": "12334",
                        "questionNumber": 1,
                        "view": Object {
                          "captured": true,
                          "html": undefined,
                          "id": "12334",
                          "picturePath": undefined,
                          "type": 1,
                        },
                      },
                    ],
                  }
                }
              />,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "grade": "3",
              "subject": "math",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "12334",
            "nodeType": "class",
            "props": Object {
              "itemData": Object {
                "passage": undefined,
                "questions": Array [
                  Object {
                    "data": Object {
                      "associatedItems": "string",
                      "commonCoreStandardsDescription": "string",
                      "depthOfKnowledge": "string",
                      "educationalDifficulty": "string",
                      "evidenceStatement": "string",
                      "itemCardViewModel": Object {
                        "bankKey": 1,
                        "claimCode": "234536",
                        "claimLabel": "Math Claim",
                        "commonCoreStandardId": "",
                        "depthOfKnowledge": "",
                        "domain": "MATH",
                        "grade": 8,
                        "gradeLabel": "grade 3",
                        "interactionTypeCode": "",
                        "interactionTypeLabel": "",
                        "isPerformanceItem": false,
                        "itemKey": 1,
                        "subjectCode": "MATH",
                        "subjectLabel": "math",
                        "target": "1-3",
                        "targetId": "23463467",
                        "title": "",
                      },
                      "rubrics": Array [
                        Object {
                          "language": "english",
                          "rubricEntries": Array [
                            Object {
                              "name": "",
                              "scorepoint": "",
                              "value": "",
                            },
                          ],
                          "samples": Array [
                            Object {
                              "maxValue": "",
                              "minValue": "",
                              "sampleResponses": Array [
                                Object {
                                  "name": "",
                                  "purpose": "",
                                  "sampleContent": "",
                                  "scorePoint": "",
                                },
                              ],
                            },
                          ],
                        },
                      ],
                      "targetDescription": "string",
                    },
                    "id": "12334",
                    "questionNumber": 1,
                    "view": Object {
                      "captured": true,
                      "html": undefined,
                      "id": "12334",
                      "picturePath": undefined,
                      "type": 1,
                    },
                  },
                ],
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "body",
      },
    ],
    "type": "html",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <head>
            <base
              href="http://example.com"
            />
            <meta
              charSet="UTF-8"
            />
            <link
              href="css/pdf.css"
              rel="stylesheet"
            />
            <link
              href="http://example.com/Shared/CSS/Universal/items.css"
              rel="stylesheet"
            />
            <link
              href="http://example.com/Projects/SBAC/css/modernItems.css"
              rel="stylesheet"
            />
          </head>,
          <body>
            <FirstPage
              grade="3"
              subject="math"
            />
            <ItemPage
              itemData={
                Object {
                  "passage": undefined,
                  "questions": Array [
                    Object {
                      "data": Object {
                        "associatedItems": "string",
                        "commonCoreStandardsDescription": "string",
                        "depthOfKnowledge": "string",
                        "educationalDifficulty": "string",
                        "evidenceStatement": "string",
                        "itemCardViewModel": Object {
                          "bankKey": 1,
                          "claimCode": "234536",
                          "claimLabel": "Math Claim",
                          "commonCoreStandardId": "",
                          "depthOfKnowledge": "",
                          "domain": "MATH",
                          "grade": 8,
                          "gradeLabel": "grade 3",
                          "interactionTypeCode": "",
                          "interactionTypeLabel": "",
                          "isPerformanceItem": false,
                          "itemKey": 1,
                          "subjectCode": "MATH",
                          "subjectLabel": "math",
                          "target": "1-3",
                          "targetId": "23463467",
                          "title": "",
                        },
                        "rubrics": Array [
                          Object {
                            "language": "english",
                            "rubricEntries": Array [
                              Object {
                                "name": "",
                                "scorepoint": "",
                                "value": "",
                              },
                            ],
                            "samples": Array [
                              Object {
                                "maxValue": "",
                                "minValue": "",
                                "sampleResponses": Array [
                                  Object {
                                    "name": "",
                                    "purpose": "",
                                    "sampleContent": "",
                                    "scorePoint": "",
                                  },
                                ],
                              },
                            ],
                          },
                        ],
                        "targetDescription": "string",
                      },
                      "id": "12334",
                      "questionNumber": 1,
                      "view": Object {
                        "captured": true,
                        "html": undefined,
                        "id": "12334",
                        "picturePath": undefined,
                        "type": 1,
                      },
                    },
                  ],
                }
              }
            />
          </body>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <base
                href="http://example.com"
              />,
              <meta
                charSet="UTF-8"
              />,
              <link
                href="css/pdf.css"
                rel="stylesheet"
              />,
              <link
                href="http://example.com/Shared/CSS/Universal/items.css"
                rel="stylesheet"
              />,
              <link
                href="http://example.com/Projects/SBAC/css/modernItems.css"
                rel="stylesheet"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "href": "http://example.com",
              },
              "ref": null,
              "rendered": null,
              "type": "base",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "charSet": "UTF-8",
              },
              "ref": null,
              "rendered": null,
              "type": "meta",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "href": "css/pdf.css",
                "rel": "stylesheet",
              },
              "ref": null,
              "rendered": null,
              "type": "link",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "href": "http://example.com/Shared/CSS/Universal/items.css",
                "rel": "stylesheet",
              },
              "ref": null,
              "rendered": null,
              "type": "link",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "href": "http://example.com/Projects/SBAC/css/modernItems.css",
                "rel": "stylesheet",
              },
              "ref": null,
              "rendered": null,
              "type": "link",
            },
          ],
          "type": "head",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <FirstPage
                grade="3"
                subject="math"
              />,
              Array [
                <ItemPage
                  itemData={
                    Object {
                      "passage": undefined,
                      "questions": Array [
                        Object {
                          "data": Object {
                            "associatedItems": "string",
                            "commonCoreStandardsDescription": "string",
                            "depthOfKnowledge": "string",
                            "educationalDifficulty": "string",
                            "evidenceStatement": "string",
                            "itemCardViewModel": Object {
                              "bankKey": 1,
                              "claimCode": "234536",
                              "claimLabel": "Math Claim",
                              "commonCoreStandardId": "",
                              "depthOfKnowledge": "",
                              "domain": "MATH",
                              "grade": 8,
                              "gradeLabel": "grade 3",
                              "interactionTypeCode": "",
                              "interactionTypeLabel": "",
                              "isPerformanceItem": false,
                              "itemKey": 1,
                              "subjectCode": "MATH",
                              "subjectLabel": "math",
                              "target": "1-3",
                              "targetId": "23463467",
                              "title": "",
                            },
                            "rubrics": Array [
                              Object {
                                "language": "english",
                                "rubricEntries": Array [
                                  Object {
                                    "name": "",
                                    "scorepoint": "",
                                    "value": "",
                                  },
                                ],
                                "samples": Array [
                                  Object {
                                    "maxValue": "",
                                    "minValue": "",
                                    "sampleResponses": Array [
                                      Object {
                                        "name": "",
                                        "purpose": "",
                                        "sampleContent": "",
                                        "scorePoint": "",
                                      },
                                    ],
                                  },
                                ],
                              },
                            ],
                            "targetDescription": "string",
                          },
                          "id": "12334",
                          "questionNumber": 1,
                          "view": Object {
                            "captured": true,
                            "html": undefined,
                            "id": "12334",
                            "picturePath": undefined,
                            "type": 1,
                          },
                        },
                      ],
                    }
                  }
                />,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "grade": "3",
                "subject": "math",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "12334",
              "nodeType": "class",
              "props": Object {
                "itemData": Object {
                  "passage": undefined,
                  "questions": Array [
                    Object {
                      "data": Object {
                        "associatedItems": "string",
                        "commonCoreStandardsDescription": "string",
                        "depthOfKnowledge": "string",
                        "educationalDifficulty": "string",
                        "evidenceStatement": "string",
                        "itemCardViewModel": Object {
                          "bankKey": 1,
                          "claimCode": "234536",
                          "claimLabel": "Math Claim",
                          "commonCoreStandardId": "",
                          "depthOfKnowledge": "",
                          "domain": "MATH",
                          "grade": 8,
                          "gradeLabel": "grade 3",
                          "interactionTypeCode": "",
                          "interactionTypeLabel": "",
                          "isPerformanceItem": false,
                          "itemKey": 1,
                          "subjectCode": "MATH",
                          "subjectLabel": "math",
                          "target": "1-3",
                          "targetId": "23463467",
                          "title": "",
                        },
                        "rubrics": Array [
                          Object {
                            "language": "english",
                            "rubricEntries": Array [
                              Object {
                                "name": "",
                                "scorepoint": "",
                                "value": "",
                              },
                            ],
                            "samples": Array [
                              Object {
                                "maxValue": "",
                                "minValue": "",
                                "sampleResponses": Array [
                                  Object {
                                    "name": "",
                                    "purpose": "",
                                    "sampleContent": "",
                                    "scorePoint": "",
                                  },
                                ],
                              },
                            ],
                          },
                        ],
                        "targetDescription": "string",
                      },
                      "id": "12334",
                      "questionNumber": 1,
                      "view": Object {
                        "captured": true,
                        "html": undefined,
                        "id": "12334",
                        "picturePath": undefined,
                        "type": 1,
                      },
                    },
                  ],
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "body",
        },
      ],
      "type": "html",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
